### PR TITLE
chore: drop union query, fair pickup and multitenant handle

### PR DESCRIPTION
**Pull request by atzoum**
# Description

Since router isolation feature is now in place, there is no longer the need for the fair pickup algorithm, i.e. `MultiTenantJobsDB` and `multitenant.Stats`. Therefore we are removing them with the following adaptations:

- `MultiTenantJobsDB#GetAllJobs` has been moved to `JobsDB#GetToProcess` and method's signature has been adapted accordingly. A future improvement shall follow which will adapt the method's current implementation and perform a single query instead of 3 separate ones.
- `processor` is now responsible for retrieving job pileup counters and populating the pending events stats during startup, what was previously done by `multitenant.Stats`

## Notion Ticket

[Link](https://www.notion.so/rudderstacks/Pipelines-Sprint-68c1c213e9b840b3a9b3826e3631e39a?p=003f3091ec0448c4b3890cee4d417b35&pm=s)

## Security

- [x] The code changed/added as part of this pull request won't create any security issues with how the software is being used.

### CODE REVIEW

File: jobsdb/jobsdb.go [hunk 1]
```diff
@@ -1771,6 +1784,74 @@ func (jd *HandleT) invalidateCacheForJobs(ds dataSetT, jobList []*JobT) {
...
+
+	unprocessed, err := jd.GetUnprocessed(ctx, params)
+	if err != nil {
+		return nil, err
+	}
+	if len(unprocessed.Jobs) > 0 {
+		unprocessedAfterJobID := unprocessed.Jobs[len(unprocessed.Jobs)-1].JobID
+		mtoken.unprocessedAfterJobID = &unprocessedAfterJobID
+	}
+	list = append(list, unprocessed.Jobs...)
```
Comment (pChondros) [inline jobsdb/jobsdb.go:1851]: [question] does it make sense to move this in the previous if block? what purpose does it serve if `len(unprocessed.Jobs) == 0`?

File: jobsdb/jobsdb.go [hunk 2]
```diff
@@ -1771,6 +1784,74 @@ func (jd *HandleT) invalidateCacheForJobs(ds dataSetT, jobList []*JobT) {
...
+
+	waiting, err := jd.GetWaiting(ctx, params)
+	if err != nil {
+		return nil, err
+	}
+	if len(waiting.Jobs) > 0 {
+		waitingAfterJobID := waiting.Jobs[len(waiting.Jobs)-1].JobID
+		mtoken.waitingAfterJobID = &waitingAfterJobID
+	}
+	list = append(list, waiting.Jobs...)
```
Comment (pChondros) [inline jobsdb/jobsdb.go:1837]: [question] does it make sense to move this in the previous if block? what purpose does it serve if `len(waiting.Jobs) == 0`?

File: router/router_test.go [hunk 3]
```diff
@@ -405,15 +398,18 @@ var _ = Describe("router", func() {
...
 
 			allJobs := append(toRetryJobsList, unprocessedJobsList...)
-			workspaceCount := map[string]int{}
-			workspaceCount[workspaceID] = len(unprocessedJobsList) + len(toRetryJobsList)
-			workspaceCountOut := workspaceCount
-
-			callGetRouterPickupJobs := mockMultitenantHandle.EXPECT().GetRouterPickupJobs(customVal["GA"], gomock.Any(), gomock.Any(), gomock.Any()).Return(workspaceCountOut).Times(1)
+			// workspaceCount := map[string]int{}
+			// workspaceCount[workspaceID] = len(unprocessedJobsList) + len(toRetryJobsList)
+			// workspaceCountOut := workspaceCount
```
Comment (pChondros) [inline router/router_test.go:403]: [question] any reason for these commented out lines?

File: router/router_test.go [hunk 4]
```diff
@@ -1038,22 +965,17 @@ var _ = Describe("router", func() {
 				},
 			}
 
-			workspaceCount := map[string]int{}
-			workspaceCount[workspaceID] = len(unprocessedJobsList) + len(toRetryJobsList)
-			workspaceCountOut := workspaceCount
+			// jobsLimit := len(unprocessedJobsList) + len(toRetryJobsList)
```
Comment (pChondros) [inline router/router_test.go:968]: [question] any reason for this commented out line?

**Review by pChondros — APPROVED** [review]

**Review by cisse21 — APPROVED** [review]

File: jobsdb/jobsdb.go (hunk 2 above)
Comment (atzoum) [reply jobsdb/jobsdb.go:1837]: if waiting.Jobs is empty this is a no-op, so it doesn't hurt either :)

**Pull request merged**